pkg/sbom/diff.go

```diff
@@ -7,6 +7,7 @@ import (
 )
 
 type NodeDiff struct {
+	NodeId    string
```

Comment:
Why do we need the NodeId here? I think most of the diff'ing scenarios will probably have different IDs for equivalent nodes, in other words one SBOM describing `packageA` will have an ID in the node while a second SBOM describing the same `packageA` will have a different node ID.

I think we should improve the APIs to find equivalent nodes instead of matching the IDs as we do now. Matching by Ids is easy but most likely will not work on most real world scenarios. 

Looking at the diff code, this ID is not really used. Should we drop it? What do you think?